<a href="https://colab.research.google.com/github/ashish-mali/EDA-On-Playstore-App-Reviews/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

<p align="center"> <img src="https://drive.google.com/uc?id=1O9138rWg9mbNXuvqhUJlYfoINFmEM5F4" width = "75%" height="365px" /> </p>


# ▶ Introdcution

<hr>

> Hi everyone, In this project we are going to analyze the <u>"Google Play Store Apps Reviews",</u> to get some insides form the data and try to analyze the <u>key factors responsible for the successful application.</u> Now, before we get the deep dive into the data to finds some insides, first we have to <u>formulate the blue print for the analysis and get to know the tools and factors associated with it.</u>



## ❇ <u>Google Play Store Overview</u>

> Google Play, often known as the Google Play Store and formerly known as Android Market, is the digital distribution service that Google operates and develops. It serves as the official app store for certified devices running over Android operating system and its variants, as well as ChromeOS, letting users to browse and download applications produced with the Android software development kit (SDK) and published through Google. In addition to offering games, music, books, movies, and television shows, Google Play also functions as a digital media shop.

>> Google play got an initial release in 2008, nearly a 14 years ago. In such a short time it's manages to impact the lives of the common people as well as business. let's see some of the stats to understand the impact of play store.
1. Android maintain the top position as a mobile operating system, accounts nearly a 70% of the market share.
2. In the year 2021, it's generated the gross revenue of 48 billion doller. (overall)
3. In the Indian context it's genreated the revenue of approx. 0.82 billion doller.
4. In India, google play enjoys wapping share of 95.1% of the total market currently.

>> As we seen from the above stats, for the business to thrive and reach to the new potential customers it is important that the business must have it's footprints in the form of an app.

### ⚙ <u>Why this analysis is important and how it impacts the overall business?</u>

> As we all know the growing power of the-smart phone market. In this todays world evey thing is avaliable in the form of an app or every business has the presence in the form of app, to engage with it's new customers and develop it's reach, which then will be converted into the business. To make this point more clear let's look some bits,

>> 1. From this analysis we get to know what are the factors affecting the app popularity.
2. how is overall category performance and top app in it, which we are targeting for the business.(In this case we take the overall view.)
3. how is competition in each sector as a download matrix. (direct impact on the number of user, will explore further.)
4. what are the matrix that determine the successful application on the play store.

> For this project we are taking the overall approch, and finding out the generic insides which then wil be helpful for the business, In terms of all it's verticals.


# ▶ A Roadmap For Analysis.

<hr>
<br>

> Befor we get deep dive into the analysis, such as exploring, cleaning and analyzing the data. It is good to have a clear understanding of the exploratory data analysis as we head on. So, each step will take us more closer to the insides we want. before we talk abut the step let's just quickly review the EDA.

## ✴ <u>What is Exploratory Data Analysis?</u>

> <b>Defination:<br> Exploratory Data Analysis refers to the critical process of <u>performing initial investigations on data so as to discover patterns,
</u>to spot anomalies,to test hypothesis and to check assumptions with the help of summary statistics and graphical representations.</b>
or<br>
<b>Exploratory Data Analysis (EDA) is an approach to analyze the data using visual techniques. It is used to discover trends, patterns, or to check assumptions with the help of statistical summary and graphical representations. </b><br>

Data scientists use exploratory data analysis (EDA) to examine data sets for patterns and abnormalities (outliers), formulate hypotheses based on the understanding of the information, and describe their key characteristics based on the domain knowledge. Visual analytics techniques are frequently employed in exploratory data analysis. In any project involving analysis of the data or in overall data science, it is a crucial stage. It assists in determining how to best manipulate data sources to achieve the desired outcomes. <br>

In order to better comprehend the data and make it easy to understand and get the quick view of the large numerical data, EDA generally employes the various charts and graphics to better understand the data in the summery format.

## ✴ <u>Six Phases Of The Data Analysis.</u>

<br>

In the real-world depending upon the organization the data analysis phases defination may subjected to chage. the the core model is always same. and each phace more or less is representative of that abstract model.<br>
<br>
So, in this project we try to replicate this real-world abstract process. The main advantage of adopting this model is that the process looks more clear and we have more focus towards goals rether than just chucking the numbers.
<br>

🔳 The following bits are clearly explain this abstract process.

> 1. <b><u>Ask</u></b> questions to make data driven decision.
>> Define problem and confirm stackholder expectation.
> 2. <b><u>Prepare</u></b> data for exploration.
>> Collect and store data for analysis.
> 3. <b><u>Process</u></b> data form dirty to clean.
>> Clean and transform data to ensure data integrity.
> 4. <b><u>Analyze</u></b> data to answer.
>> Use data analysis tools to draw conclusions.
> 5. <b><u>Share</u></b> data through the art of visulization.
>> Interpret and communicate result for data-driven decision.
> 6. <b><u>Act</u></b> on the insides to make real-time change.
>> Put your insides to work in order to solve problem.

### ✳ <u>Six phases of data analysis in the context of this project.</u>

✅ <b><u>Ask</u></b> :

> Asking the right question is the most important part of the whole data analysis process. The quality of your query desides how impactful your insides are. Gernerally in this phase we setup a team-meet with the stackholder or with higher management to figure out the problem or what they wanted to analyze.<br>
>> For this project we have to brainsttom a bit on a given data-set and outline some of the questions which then will be converted into the insides.
<br>

✅ <b><u>Prepare</b></u> :

> It's all about collecting and storing the data. Preparing the data-set for the exploration is the main task for the this process. This step is important when we have to collect the data from the different third-party vendors to answer our question. In this process we also have to figure-out what type of data we required so as to answer our question.
following are the some common question we asked during this process.<br>
>> 🇶. Is data is enough?<br>
>> 🇶. Is internal data is enough to answer, or we required some second-party or even third-party data, if so what is the cost of it, is it worth?.<br>

After figuring out the data, we have to collect and store it in the databases for further analysis.
<br>
>> For this project we already given two data file in the comma seperated values format(.csv). more on this when we go to this section.
<br>

✅ <b><u>Process</b></u> :

> This step is the basic of our analysis, Or in other words we can say that this is the hart and soul of the process. we more frequently used the ***<u>Data Cleaning</u>*** term for it. more on this in the process section.
<br>
>> For this project we must have to clean the data file that are given in order to draw the conclusion.
<br>

✅ <b><u>Analyze</b></u> :

> After completing the process phace now our data is ready to analyze. we have enormous tools avaliable to analyze the data, which tools to used is totally depends upon the organization we are worked in 
<br>
>> For this project we use, <u>***python***</u>, <u>***numpy***</u>, <u>***pandas***</u>, <u>***matplotlib***</u> and <u>***seaborn***</u>.
we further use <u>***microsoft word***</u> for generating the reports.
<br>

✅ <b><u>Share</b></u> :

> Sharing our finding to the stackholders or the person with higher authority is the sole purpose of this step. some time we have to provide static reports for example <u>***Summery of project***</u> or dynamic report aka interactive or live reports made with <u>***tableau***</u>.
<br>
>> For this project we make the static summery report highlighing the key finding.
<br>

✅ <b><u>Act<b></u> :

> Now, its time to see our insides into the actions and make the difference in real time. In this phase it's more about the stackeholder or subject matter export to how they wanted to implemented this changes.
<br>
>> For this project we define the <u>***Conclusion***</u> part within it. so that we have to clear what type of the finding we get after doing this analysis.

# ▶ The Essentials.

> As we done the analysis in the "google colab" it's important to follow the some steps those are :<br>

>> 1. Mounting the drive into the colab notebook
2. Importing the essential labary for the analysis
3. Importing the data to explore
4. Setting-up the initial setup
<br>

In [1]:
# mounting the drive to colab notebook.
# this the location where we keep our related file as a single entity.
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
#import librarys

# numpy for linear algebric calculation
import numpy as np
# pandas for data wrangling
import pandas as pd 
# matplotlib for ploting the charts.
import matplotlib.pyplot as plt
# visualization tool
import seaborn as sns 
# importing the datatime module from python
from datetime import datetime
# plotly
import plotly 
plotly.offline.init_notebook_mode(connected=True)
import plotly.graph_objs as go
import warnings
#sns.set(font_scale=1.5)
warnings.filterwarnings("ignore")

In [3]:
# giving the path of directory where all project files are located.
working_dir = "/content/drive/MyDrive/AlmaBetter/Module 1/Play Store App Review Analysis - Ashish Mali/Data_Files/"

# this is the file path for "Play_Store_Data.csv".
ps_data_file_path = (working_dir + "/Play_Store_Data.csv")

# this is the file path for "User_Reviews_Data.csv"
usr_review_data_file_path = (working_dir + "/User_Reviews_Data.csv")

In [4]:
# creating the dataframe for the playstore.
playstore_df = pd.read_csv(ps_data_file_path)

In [5]:
# printing the first five observation and their characteristic.
playstore_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [6]:
# creating the data frame for the user review
user_review_df = pd.read_csv(usr_review_data_file_path)

In [7]:
# printing the first five observation and their characteristic.
user_review_df.head()

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   
3  10 Best Foods for You         Works great especially going grocery store   
4  10 Best Foods for You                                       Best idea us   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN  
3  Positive                0.40                0.875000  
4  Positive                1.00                0.300000

In [8]:
# just printing how many rows and columns we have for analysis for playstore dataframe.
print(f"For playstore_df we have {playstore_df.shape[0]} records and {playstore_df.shape[1]} attributes.") 

# just printing how many rows and columns we have for analysis for user review dataframe.
print(f"For user_review_df we have {user_review_df.shape[0]} records and {user_review_df.shape[1]} attributes.") 

For playstore_df we have 10841 records and 13 attributes.
For user_review_df we have 64295 records and 5 attributes.


## ✳ <u>Creating the copys of the dataframes.</u>

> As we have our two dataframe for analysis, it's good idea to create copy of it for the analysis. because we done a several transformation and manuplication on the dataframe to extract the insides and the corelations among the many attributes,it may cause the loss of original data a bit. especially after the data cleaning. so, for that case we must have a backup copies of the data.

In [9]:
# creating the copy of the playstore_df.
ps_df = playstore_df.copy()

ps_df.head(3)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   

  Reviews  Size    Installs  Type Price Content Rating  \
0     159   19M     10,000+  Free     0       Everyone   
1     967   14M    500,000+  Free     0       Everyone   
2   87510  8.7M  5,000,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up  
2               Art & Design    August 1, 2018       1.2.4  4.0.3 and up

In [10]:
# creating the copy of the user_review_df.
ur_df = user_review_df.copy()

ur_df.head(3)

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN

# ▶ Ask question to make data-driven decisions.
<hr>

> As we have data in our hand we have to bit explore this data to get to know what type it is and what it signifies and what type of conclusion we can drow from it. seem to have just reversed process as it involving the data at first phace itself, we mark it as an exception for this analysis and get to the business.

## ✴ <u>Problem Statements.</u>

> The main purpose of this analysis is to analyze and find out the key factors responsible for the success of the application.<br>
So, this is the broad term we have for analysis, the success of the application is depends upon the various factors, one such factor is user engagement. like that we have several other things to answer and find out the required things we must do for the successful application.<br>
>> 1. Find out the number of application per categorie.?
2. For each categorie find the top 3 apps based on the various factors.?
3. Among all the apps which one has a highest number of installs, query all it's attributes?
4. Find out the percentage of free apps over a paid apps?
5. Find the top categories on play store based on the various factors.?
6. Find out the effect of application rating over it's user engagement.?
7. Does the size of the application matter, what's it's effect on rating and user engagement.?
8. Find the total number of application by various Genres.?
9. Find out the top apps with respect to free and paid, analyze it's rating and number of installs?
10. Find out the relationship between rating and reviews.?
11. What is the sentiment subjectivity.
12. Is subjectivity and polarity proportional to each other?
13. What is the percentage of review sentiments?
14. How is sentiment polarity varying for paid and free apps?
15. How Content Rating affect over the App?
16. Does Last Update date has an effects on rating?
17. How is the Distribution of app update over the fixed period.?
18. Is content rating has an impact on reviews.?
19. Find the percentage of reviews over install, discuss this case.?
20. what is the impact of android version attribute over the install.?


# ▶ Prepare Data For Exploration.
<hr>
<br>

> Generally, in this step we have to look for the data that answer our questions. and featch and store that data for further analysis. In this step we have to look at various data sources such as,
<br>
>> 1. Internal data.
2. External data.
3. Open data repo.
4. First party data.
4. Second party data.
5. Third party data
<br>

Combine all the data which is required to answer our burning question is import. It is also happen we the data we are looking for is not avaliable at all, so we have to find out the approch to get it at first place. for example conducting surveys.
<br>

>> For this project we have the data in hand and it is sufficient to answer our all the question so no need to collect more information. next we have to store this data which handed over as <code>.csv</code> files. for this we choose the <code>google drive (cloud solution)</code> as place to store it.
<br>

Now, <b><u>let's Explore the data and define each attribute.</u></b>

In [11]:
# let's display the data of the play store.
head_tail_combo = pd.concat([ps_df.head(), ps_df.tail()])

In [12]:
# showing the dataframe consist of top and bottom 5 recoreds in one frame.
head_tail_combo

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device

> From the above metadata we get to the which attributes having the null values and the total number of attributes present in the dataframe. we also get to know that the data types of each attribute. also some important information about the memory usage.

In [13]:
# ones again fatching the total number of records and it's attributes.
print(f"For playstore dataframe we have {ps_df.shape[0]} records and {ps_df.shape[1]} attributes.")

For playstore dataframe we have 10841 records and 13 attributes.


In [14]:
# fetching the names of all 13 attributes of the dataframe in form of list.
print(list(ps_df.columns))

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


In [15]:
# let's look some metadata related to our data
print(ps_df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB
None


In [16]:
# fetching total number of null value present in each attribute.
print(ps_df.isnull().sum())

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64


In [17]:
# let's get some stastical information.
ps_df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000


## ✴ <u>Synopsis</u>

> We just wrap-up the data exploration, following are the bitwise finding we get after the process.<br>



*   The total number of observation are 10841.
*   Each observation has 13 attributes
*   There are in total 5 attributes that hold the null values.
*   Among these 13 attributes <code>"Rating"</code> attribute has highest number of null values which accouts to 1474.
*   The lowest number of null values hold by <code>"Type"</code> and <code>"Content Rating"</code> attribute, account to 1 null value each.
*   The other attribute such as <code>"Current var"</code> and <code>"Android ver"</code> each take 8 and 3 null values respectively.
*   All the attribute except the <code>"Rating"</code> attribute hold <code>"object"</code> data type.
*   The <code>"Rating"</code> attribute has an data type of <code>"float64"</code>, the only attribute which holds numerical data.
*   The data type of the follwing attrubutes must changed to <code>int</code> type in order to perform analysis :
> 'Rating', 'Reviews', 'Size', 'Installs', 'Price', 'Content Rating', 'Current Ver', 'Android Ver'<br>

*   The <code>'Last Updated'</code> column data type need to be changed to <code>date</code> type.
*   In the dataframe each record/row represent the application.
*   Iet's shortly describe what each column/attribute represent :<br>
> 1. <code><b>App</b></code> - It tells us about the name of the application with a short description (optional).
> 2. <code><b>Category</code></b> - It gives the category to the app.
> 3. <code><b>Rating</code></b> - It contains the average rating of the app out of 5. (also known as Likert scale)
> 4. <code><b>Reviews</code></b> - It tells us about the total number of users who have given a review for the application.
> 5. <code><b>Size</code></b> - It tells us about the memory space required for application in mega-bits.
> 6. <code><b>Installs</code></b> - It tells us about the total number of installs/downloads for an application.
> 7. <code><b>Type</code></b> - It talk about wheter app is free or paid, contain only two possible values.
> 8. <code><b>Price</code></b> - It gives the price for payable app to use, for the free app it takes the value 0.
> 9. <code><b>Content Rating</code></b> - It states the minimum age limit to use the application.
> 10. <code><b>Genres</code></b> - It tells us about the various other categories to which an application can belong.
> 11. <code><b>Last Updated</code></b> - It tells us about the when the application was last updated.
> 12. <code><b>Current Ver</code></b> - It tells us about the current version of the application.
> 13. <code><b>Android Ver</code></b> - It tells from which android version onwords, the application is recommended to install and perform smooth operation. 